devserver_lib/src/lib.rs

```diff
@@ -17,6 +16,26 @@ use std::thread;
 #[cfg(feature = "reload")]
 mod reload;
 
+struct FileProxy {
+    path: String,
+    extension: Option<String>
+}
+impl FileProxy {
+    fn new(path: String, extension: Option<String>) -> Self {
+        let owned_ext = match extension {
+            Some(ext) => Some(ext),
+            None => None,
+        };
+        FileProxy { path: path, extension: owned_ext }
```

Comment:
Based on the code you have here the match statement isn't doing anything.

This is a bit weird, but in Rust if the name of a variable is the same as the thing you're assigning in the struct you can initialize the struct like this:

```rust
FileProxy { path, extension }
 ```
instead of:

```rust
FileProxy { path: path, extension: extension }
```

So in this case the function can be reduced to:
```rust
fn new(path: String, extension: Option<String>) -> Self {
    FileProxy {path, extension} // This can also be written as `Self {path, extension}`
}
```